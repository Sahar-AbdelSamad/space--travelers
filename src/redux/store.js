import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer, { fetchRockets } from './rockets/rocketsSlice';
import missionsReducer, { fetchMissions } from './missions/missionsSlice';
import mobileMenuReducer from './mobileMenu/mobileMenuSlice';

export const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
    mobileMenu: mobileMenuReducer,
  },
});

store.dispatch(fetchRockets());
store.dispatch(fetchMissions());

export default store;
