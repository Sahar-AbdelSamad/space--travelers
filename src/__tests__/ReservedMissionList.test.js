import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import ReservedMissionsList from '../components/ReservedMissionList';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ReservedMissionsList />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
