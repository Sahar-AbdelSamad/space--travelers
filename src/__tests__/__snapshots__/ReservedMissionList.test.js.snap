// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <p>
    no missions joined
  </p>,
  <table
    className="profile-table"
  >
    <tbody />
  </table>,
]
`;
